### PR TITLE
New icons guidelines

### PR DESCRIPTION
**Pull request by maximedegreve**
Currently, our icon documentation is located within the Octicons library, which can be overlooked when browsing through our guidelines. This pull request aims to centralize all of our documentation by moving the icon page to our guidelines and improving it.

Our ultimate goal is to centralize all of our documentation for easier access. However, I have intentionally left the [Octicons design guidelines](https://primer.style/octicons/guidelines/design) in place for now, as I believe it is still the best place for information on creating new icons. I have added a section called `Creating New Icons` to direct users to this information.

In the past year, we have received numerous questions about using Octicons as selectable icons for users. To address this, I have added a section called `System vs User-Selectable` to compile past discussions on this topic.

I've also simplified our icon pairing section as those guidelines were very inconsistent and confusing.

The do and don't section are also now improved.

The `Special` section also adds more strict documentation on using special icons.

**Preview:** 
https://primer-a9164a06d1-26441320.drafts.github.io/components/icon

**Screenshot:**
Screenshot 2023-01-03 at 15 51 48

### CODE REVIEW

File: content/components/icon.mdx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+## Library
+
+At Primer, we utilize the [Octicons](/octicons/) icon library. This package is available in [Ruby](/octicons/packages/ruby), [Jekyll](/octicons/packages/jekyll), [JavaScript](/octicons/packages/javascript), [React](/octicons/packages/react) and [Styled System](/octicons/packages/styled-system).
+
+## Usage
+
+Icons are most appropriate for representing simple, recognizable concepts or actions. When choosing icons for your design system, consider the following guidelines:
+
+1. Where possible use icons to supplement text, rather than replacing it.
+2. Be sure that the meaning of icons is clear when used without accompanying text..
```
Comment (lukasoppermann) [inline content/components/icon.mdx:26]:
```suggestion
2. Make sure that the meaning of icons is clear when used without accompanying text.
```

File: content/components/icon.mdx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+3. Choose icons that are easily recognizable and accurately depict the intended concept or action.
```
Comment (lukasoppermann) [inline content/components/icon.mdx:27]:
```suggestion
3. Choose icons that are easy to recognize and accurately depict the concept or action.
```

File: content/components/icon.mdx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+4. Maintain consistency in the use of icons across your product or brand to establish a cohesive visual language.
```
Comment (lukasoppermann) [inline content/components/icon.mdx:28]:
```suggestion
4. Maintain consistency in the use of icons across your product or brand.
```
I feel the last bit makes it less clear what you mean.

Also does `product or brand` make sense here or would it just be `product`?

File: content/components/icon.mdx [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+5. Avoid using icons for complex or abstract concepts that may be confusing to some users.
+6. Adhere to predetermined icon sizes to ensure legibility and ease of recognition.
+7. Test the effectiveness of your icons with users to ensure that they are understood and useful in navigating the interface.
+
+## Color
+
+To ensure that icons are effective and legible in different color modes, it is recommended to use our [functional foreground colors](/foundations/color#foregrounds). This helps to maintain consistency and ensures that the icons convey the intended meaning to users.
```
Comment (lukasoppermann) [inline content/components/icon.mdx:35]:
```suggestion
To ensure that icons are effective and legible in different color modes, it is recommended to use our [functional foreground colors](/foundations/color#foregrounds).
```
I feel you can get rid of the second sentence without losing any info.

File: content/components/icon.mdx [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+
+System icons are used to represent core actions and concepts within the user interface. These icons are typically used consistently throughout the product and are an important part of our visual language. Examples of system icons might include a [trash](/octicons/trash-16) icon for deleting items, or a [plus-circle](/octicons/plus-circle-16) icon for adding new items.
+
+User-selectable icons, on the other hand, are used to allow users to customize their experience or make a personal connection with the product. User-selectable icons are typically less central to the functionality of the user interface and may be changed by the user as desired.
+
+At GitHub we currently use Emojies as user-selectable icons as they are widely recognized and understood, making them an effective way to convey emotions or personality. One benefit of using emojis as user-selectable icons is that they are visually distinct from the system icons used in the design system. This helps users to easily understand that the emojis are optional and can be selected by the user, rather than being a part of the core functionality of the user interface.
```
Comment (josepmartins) [inline content/components/icon.mdx:126]:
```suggestion
At GitHub we currently use emojis as user-selectable icons as they are widely recognized and understood, making them an effective way to convey emotions or personality. One benefit of using emojis as user-selectable icons is that they are visually distinct from the system icons used in the design system. This helps users to easily understand that the emojis are optional and can be selected by the user, rather than being a part of the core functionality of the user interface.
```

File: content/components/icon.mdx [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+4. Maintain consistency in the use of icons across your product or brand to establish a cohesive visual language.
+5. Avoid using icons for complex or abstract concepts that may be confusing to some users.
+6. Adhere to predetermined icon sizes to ensure legibility and ease of recognition.
+7. Test the effectiveness of your icons with users to ensure that they are understood and useful in navigating the interface.
+
+## Color
+
+To ensure that icons are effective and legible in different color modes, it is recommended to use our [functional foreground colors](/foundations/color#foregrounds). This helps to maintain consistency and ensures that the icons convey the intended meaning to users.
+
+Many of our components are designed to automatically set the color of icons to a predetermined color. For example, flash banners are configured to set the icon to our [fg.danger](/foundations/color#foregrounds) functional color by default.
```
Comment (lukasoppermann) [inline content/components/icon.mdx:37]: This is not quite accurate I think. `fg.danger` is only used for the flash banner in the alert type. So maybe you should say something like this:

```suggestion
Many of our components are designed to automatically set the color of icons to a predetermined color. For example, flash banners are configured to set the icon to the matching [role color](/foundations/color#foregrounds) by default.
```

File: content/components/icon.mdx [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+
+<img
+  width="960"
+  alt="A flash banner showing a warning icon in orange with the text 'Your password was reset recently.'"
+  src="https://user-images.githubusercontent.com/980622/210349549-e7444c43-bb52-4a84-b00e-13962a22377a.png"
+/>
+
+In some cases, certain icons should always be paired with specific colors in order to effectively convey their intended meaning and provide a consistent user experience. Here are a few examples:
```
Comment (lukasoppermann) [inline content/components/icon.mdx:45]: This may be easier to read (or not, not 100% sure 😅)

```suggestion
Some icons should always be shown in a specific colors when used within a specific context. This helps to effectively convey their intended meaning and provide a consistent user experience. Here are a few examples:
```

File: content/components/icon.mdx [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+    />
+    <Caption>Don't resize icons.</Caption>
+  </Dont>
+</DoDontContainer>
+
+## System vs user-selectable
+
+There are two types of icons: `system icons` and `user-selectable` icons.
+
+System icons are used to represent core actions and concepts within the user interface. These icons are typically used consistently throughout the product and are an important part of our visual language. Examples of system icons might include a [trash](/octicons/trash-16) icon for deleting items, or a [plus-circle](/octicons/plus-circle-16) icon for adding new items.
```
Comment (lukasoppermann) [inline content/components/icon.mdx:122]:
```suggestion
`System icons` are used to represent core actions and concepts within the user interface. These icons are typically used consistently throughout the product and are an important part of our visual language. Examples of system icons might include a [trash](/octicons/trash-16) icon for deleting items, or a [plus-circle](/octicons/plus-circle-16) icon for adding new items.
```

File: content/components/icon.mdx [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+
+User-selectable icons, on the other hand, are used to allow users to customize their experience or make a personal connection with the product. User-selectable icons are typically less central to the functionality of the user interface and may be changed by the user as desired.
+
+At GitHub we currently use Emojies as user-selectable icons as they are widely recognized and understood, making them an effective way to convey emotions or personality. One benefit of using emojis as user-selectable icons is that they are visually distinct from the system icons used in the design system. This helps users to easily understand that the emojis are optional and can be selected by the user, rather than being a part of the core functionality of the user interface.
+
+Allowing users to pick [Octicons](/octicons/) (system icons) is strongly discouraged and could create a confusing experience when they are used in the wrong context. For example a user could pick a [issue-closed](https://primer.style/octicons/issue-closed-16) icon and use the label `In progress`.
```
Comment (josepmartins) [inline content/components/icon.mdx:128]: I'd rephrase this a bit to keep using the `user-selectable` term
```suggestion
Using [Octicons](/octicons/) as user-selectable icons is strongly discouraged as that could create a confusing experience when used in the wrong context. For example, a user could pick a [issue-closed](https://primer.style/octicons/issue-closed-16) icon and use the label `In progress`.
```

File: content/components/icon.mdx [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+  </Dont>
+</DoDontContainer>
+
+## System vs user-selectable
+
+There are two types of icons: `system icons` and `user-selectable` icons.
+
+System icons are used to represent core actions and concepts within the user interface. These icons are typically used consistently throughout the product and are an important part of our visual language. Examples of system icons might include a [trash](/octicons/trash-16) icon for deleting items, or a [plus-circle](/octicons/plus-circle-16) icon for adding new items.
+
+User-selectable icons, on the other hand, are used to allow users to customize their experience or make a personal connection with the product. User-selectable icons are typically less central to the functionality of the user interface and may be changed by the user as desired.
```
Comment (lukasoppermann) [inline content/components/icon.mdx:124]:
```suggestion
`User-selectable` icons, on the other hand, are used to allow users to customize their experience or make a personal connection with the product. `User-selectable` icons are typically less central to the functionality of the user interface and may be changed by the user as desired.
```

File: content/components/icon.mdx [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+
+<DoDontContainer>
+  <Do>
+    <img
+      src="https://user-images.githubusercontent.com/980622/209167301-278f8dff-70a1-4bc3-9ad9-335fb365bd67.png"
+      role="presentation"
+      width="456"
+      alt="A navigation side bar with multiple items including 'To do', 'Learning' and ''Ready to go. Each item has a emoji assigned and is displayed on the left of the text."
+    />
+    <Caption>Use Emojies as user-selectable icons.</Caption>
```
Comment (josepmartins) [inline content/components/icon.mdx:138]:
```suggestion
    <Caption>Use emojis as user-selectable icons.</Caption>
```

File: content/components/icon.mdx (hunk 5 above)
Comment (lukasoppermann) [inline content/components/icon.mdx:126]:
```suggestion
At GitHub we currently use emojies as `user-selectable` icons as they are widely recognized and understood, making them an effective way to convey emotions or personality. One benefit of using emojis as user-selectable icons is that they are visually distinct from the system icons used in the design system. This helps users to easily understand that the emojis are optional and can be selected by the user, rather than being a part of the core functionality of the user interface.
```

File: content/components/icon.mdx [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+      width="456"
+      alt="A pencil icon scaled up from 16px to 24px demonstrating how the stroke increased while upscaling. Another pencil icon is scaled down from 24px to 16px and demonstrates how the stroke became too thin."
+    />
+    <Caption>Don't resize icons.</Caption>
+  </Dont>
+</DoDontContainer>
+
+## System vs user-selectable
+
+There are two types of icons: `system icons` and `user-selectable` icons.
```
Comment (lukasoppermann) [inline content/components/icon.mdx:120]: Maybe the `icons` part should be in the backticks for `user-selectable` as well. So `user-selectable icons`?
This would than also apply to my below comments.

File: content/components/icon.mdx [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
+      src="https://user-images.githubusercontent.com/980622/209167308-0c8cca85-5d78-4c60-b69b-13f5afbf0254.png"
+      role="presentation"
+      width="456"
+      alt="A navigation side bar with multiple items including 'To do', 'Learning' and ''Ready to go. Each item has a octicon assigned in a different icon and is displayed on the left of the text."
+    />
+    <Caption>Don't use Octicons for user-selectable icons.</Caption>
+  </Dont>
+</DoDontContainer>
+
+Note that internally we had multiple discussions about introducing user-selectable octicons but no specific work or explorations have been started on this yet. This work initially would involve creating a set of octicons that look visually different from our existing style.
```
Comment (josepmartins) [inline content/components/icon.mdx:151]: I have mixed feelings about this part. Even if this is true, should we add a note like that? This might sound like emojis are a temporary/quick solution 🤔.

```suggestion
Note that internally we had multiple discussions about introducing user-selectable octicons but no specific work or explorations have been started on this yet. This work initially would involve creating a set of octicons that look visually different from our existing style.
```

File: content/components/icon.mdx [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+      alt="A navigation side bar with multiple items including 'To do', 'Learning' and ''Ready to go. Each item has a emoji assigned and is displayed on the left of the text."
+    />
+    <Caption>Use Emojies as user-selectable icons.</Caption>
+  </Do>
+  <Dont>
+    <img
+      src="https://user-images.githubusercontent.com/980622/209167308-0c8cca85-5d78-4c60-b69b-13f5afbf0254.png"
+      role="presentation"
+      width="456"
+      alt="A navigation side bar with multiple items including 'To do', 'Learning' and ''Ready to go. Each item has a octicon assigned in a different icon and is displayed on the left of the text."
```
Comment (lukasoppermann) [inline content/components/icon.mdx:145]:
```suggestion
      alt="A navigation side bar with multiple items including 'To do', 'Learning' and ''Ready to go. Each item has an octicon assigned that is displayed on the left of the text."
```

File: content/components/icon.mdx (hunk 13 above)
Comment (lukasoppermann) [inline content/components/icon.mdx:151]:
```suggestion
Note that internally we had multiple discussions about introducing user-selectable Octicons but no specific work or explorations have been started on this yet. This work initially would involve creating a set of Octicons that look visually different from our existing style.
```

Comment (lukasoppermann) [reply content/components/icon.mdx:151]: Should `Octicons` be linked here as well?

**Comment by lukasoppermann** [comment]
This is great @maximedegreve 🙏 so good to have this now.

File: content/components/icon.mdx [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+title: Icon
+react: https://primer.style/octicons/packages/react
+rails: https://primer.style/octicons/packages/ruby
+figma: https://www.figma.com/file/1ljgTFkT5NKNRfq5hw07JQ/Octicons?node-id=1%3A2&t=Lz5afoq1bbWrD9Yt-1
+description: Our icons are located in a separate library called Octicons, which is available in various implementations including React, Figma, Rails, and Styled System.
+---
+
+import {Box, Text} from '@primer/react'
+
+<img
```
Comment (josepmartins) [inline content/components/icon.mdx:11]: ❤️

File: content/components/icon.mdx [hunk 16]
```diff
@@ -0,0 +1,216 @@
...
+| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: | ------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------ |
+| <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 01-1.146 1.954H1.33A1.313 1.313 0 01.183 9.058zM7 7V3H5v4zm-1 3a1 1 0 100-2 1 1 0 000 2z"></path></svg> | [`alert-fill`](https://primer.style/octicons/alert-fill-12)               | For cautionary messaging or to inform the user that an action requires attention                                               |
```
Comment (josepmartins) [inline content/components/icon.mdx:88]: Removing ` to be consistent with the rest of the icon-link columns

```suggestion
| <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M4.855.708c.5-.896 1.79-.896 2.29 0l4.675 8.351a1.312 1.312 0 01-1.146 1.954H1.33A1.313 1.313 0 01.183 9.058zM7 7V3H5v4zm-1 3a1 1 0 100-2 1 1 0 000 2z"></path></svg> | [alert-fill](https://primer.style/octicons/alert-fill-12)               | For cautionary messaging or to inform the user that an action requires attention                                               |
```

File: content/components/icon.mdx [hunk 17]
```diff
@@ -0,0 +1,216 @@
...
+|                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M6 0a6 6 0 110 12A6 6 0 016 0zm-.705 8.737L9.63 4.403 8.392 3.166 5.295 6.263l-1.7-1.702L2.356 5.8l2.938 2.938z"></path></svg>                     | [`check-circle-fill`](https://primer.style/octicons/check-circle-fill-12) | For positive messaging to inform the user that an action is successful, complete, or that they may continue through a workflow |
```
Comment (josepmartins) [inline content/components/icon.mdx:89]:
```suggestion
|                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M6 0a6 6 0 110 12A6 6 0 016 0zm-.705 8.737L9.63 4.403 8.392 3.166 5.295 6.263l-1.7-1.702L2.356 5.8l2.938 2.938z"></path></svg>                     | [check-circle-fill](https://primer.style/octicons/check-circle-fill-12) | For positive messaging to inform the user that an action is successful, complete, or that they may continue through a workflow |
```

File: content/components/icon.mdx [hunk 18]
```diff
@@ -0,0 +1,216 @@
...
+|                                                       <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M6 0a6 6 0 110 12A6 6 0 016 0zm3 5H3v2h6z"></path></svg>                                                        | [`no-entry-fill`](https://primer.style/octicons/no-entry-fill-12)         | Indicate an ending or that the user is blocked and cannot continue                                                             |
```
Comment (josepmartins) [inline content/components/icon.mdx:90]:
```suggestion
|                                                       <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M6 0a6 6 0 110 12A6 6 0 016 0zm3 5H3v2h6z"></path></svg>                                                        | [no-entry-fill](https://primer.style/octicons/no-entry-fill-12)         | Indicate an ending or that the user is blocked and cannot continue                                                             |
```

File: content/components/icon.mdx [hunk 19]
```diff
@@ -0,0 +1,216 @@
...
+|     <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M1.757 10.243a6.001 6.001 0 118.488-8.486 6.001 6.001 0 01-8.488 8.486zM6 4.763l-2-2L2.763 4l2 2-2 2L4 9.237l2-2 2 2L9.237 8l-2-2 2-2L8 2.763z"></path></svg>     | [`x-circle-fill`](https://primer.style/octicons/x-circle-fill-12)         | For negative messaging to inform the user that an error has occurred as a result of an action or an action is unavailable      |
```
Comment (josepmartins) [inline content/components/icon.mdx:91]:
```suggestion
|     <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M1.757 10.243a6.001 6.001 0 118.488-8.486 6.001 6.001 0 01-8.488 8.486zM6 4.763l-2-2L2.763 4l2 2-2 2L4 9.237l2-2 2 2L9.237 8l-2-2 2-2L8 2.763z"></path></svg>     | [x-circle-fill](https://primer.style/octicons/x-circle-fill-12)         | For negative messaging to inform the user that an error has occurred as a result of an action or an action is unavailable      |
```

File: content/components/icon.mdx [hunk 20]
```diff
@@ -0,0 +1,216 @@
...
+|       <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M6 8.8c-.2 0-.4-.1-.5-.2L2.2 5.3c-.3-.3-.3-.8 0-1.1.3-.3.8-.3 1.1 0L6 6.9l2.7-2.7c.3-.3.8-.3 1.1 0 .3.3.3.8 0 1.1L6.6 8.5c-.2.2-.4.3-.6.3z"></path></svg>       | [`chevron-down`](https://primer.style/octicons/chevron-down-12)           | Indicates that a collapsible section is currently open                                                                         |
```
Comment (josepmartins) [inline content/components/icon.mdx:92]:
```suggestion
|       <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M6 8.8c-.2 0-.4-.1-.5-.2L2.2 5.3c-.3-.3-.3-.8 0-1.1.3-.3.8-.3 1.1 0L6 6.9l2.7-2.7c.3-.3.8-.3 1.1 0 .3.3.3.8 0 1.1L6.6 8.5c-.2.2-.4.3-.6.3z"></path></svg>       | [chevron-down](https://primer.style/octicons/chevron-down-12)           | Indicates that a collapsible section is currently open                                                                         |
```

File: content/components/icon.mdx [hunk 21]
```diff
@@ -0,0 +1,216 @@
...
+|      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M4.7 10c-.2 0-.4-.1-.5-.2-.3-.3-.3-.8 0-1.1L6.9 6 4.2 3.3c-.3-.3-.3-.8 0-1.1.3-.3.8-.3 1.1 0l3.3 3.2c.3.3.3.8 0 1.1L5.3 9.7c-.2.2-.4.3-.6.3z"></path></svg>      | [`chevron-right`](https://primer.style/octicons/chevron-right-12)         | Indicates that a collapsible section is currently closed                                                                       |
```
Comment (josepmartins) [inline content/components/icon.mdx:93]:
```suggestion
|      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 12 12" width="12" height="12"><path d="M4.7 10c-.2 0-.4-.1-.5-.2-.3-.3-.3-.8 0-1.1L6.9 6 4.2 3.3c-.3-.3-.3-.8 0-1.1.3-.3.8-.3 1.1 0l3.3 3.2c.3.3.3.8 0 1.1L5.3 9.7c-.2.2-.4.3-.6.3z"></path></svg>      | [chevron-right](https://primer.style/octicons/chevron-right-12)         | Indicates that a collapsible section is currently closed                                                                       |
```

**Review by josepmartins — COMMENTED** [review]
This is awesome @maximedegreve! 

As a side note, the new IA that @emilybrick is working on adds `Icons` as part of the`/foundations` section and not as /components, which I think it makes more sense as a general guidelines for icons.

File: content/components/icon.mdx (hunk 13 above)
Comment (maximedegreve) [reply content/components/icon.mdx]: You're right. Removed it for now.

File: content/components/icon.mdx (hunk 12 above)
Comment (maximedegreve) [reply content/components/icon.mdx]: Good spot I've changed it to `system` to be consistent with the title.

File: content/components/icon.mdx [hunk 22]
```diff
@@ -0,0 +1,214 @@
+---
+title: Icon
+react: https://primer.style/octicons/packages/react
+rails: https://primer.style/octicons/packages/ruby
+figma: https://www.figma.com/file/1ljgTFkT5NKNRfq5hw07JQ/Octicons?node-id=1%3A2&t=Lz5afoq1bbWrD9Yt-1
+description: Our icons are located in a separate library called Octicons, which is available in various implementations including React, Figma, Rails, and Styled System.
```
Comment (emilybrick) [inline content/components/icon.mdx:6]: It feels a little strange to start off by saying where the Icons are located (especially if we're not linking to it here). It looks like you've got it covered in the text below. 

```suggestion
description: Icons at GitHub are called Octicons, which are available in various implementations including React, Figma, Rails, and Styled System. 
```

File: content/components/icon.mdx [hunk 23]
```diff
@@ -0,0 +1,214 @@
...
+  width="960"
+  alt="An image of multiple icons placed in a grid"
+  src="https://user-images.githubusercontent.com/980622/210349372-9392f4b3-4902-4fd3-ab6f-fbd3587f208a.png"
+/>
+
+## Library
+
+At Primer, we utilize the [Octicons](/octicons/) icon library. This package is available in [Ruby](/octicons/packages/ruby), [Jekyll](/octicons/packages/jekyll), [JavaScript](/octicons/packages/javascript), [React](/octicons/packages/react) and [Styled System](/octicons/packages/styled-system).
+
+## Usage
```
Comment (emilybrick) [inline content/components/icon.mdx:21]: love this!

File: content/components/icon.mdx [hunk 24]
```diff
@@ -0,0 +1,214 @@
...
+|                                                                  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M3.72 3.72a.75.75 0 011.06 0L8 6.94l3.22-3.22a.75.75 0 111.06 1.06L9.06 8l3.22 3.22a.75.75 0 11-1.06 1.06L8 9.06l-3.22 3.22a.75.75 0 01-1.06-1.06L6.94 8 3.72 4.78a.75.75 0 010-1.06z" fill="#D73A49"></path></svg>                                                                   | [x](/octicons/x-16)         | [fg.danger](/foundations/color#foregrounds)    | Error, danger, or negative result       |
+| <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" width="16" height="16"><path fill-rule="evenodd" d="M8.22 1.754a.25.25 0 00-.44 0L1.698 13.132a.25.25 0 00.22.368h12.164a.25.25 0 00.22-.368L8.22 1.754zm-1.763-.707c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0114.082 15H1.918a1.75 1.75 0 01-1.543-2.575L6.457 1.047zM9 11a1 1 0 11-2 0 1 1 0 012 0zm-.25-5.25a.75.75 0 00-1.5 0v2.5a.75.75 0 001.5 0v-2.5z" fill="#B08800"></path></svg> | [alert](/octicons/alert-16) | [fg.attention](/foundations/color#foregrounds) | Warning                                 |
+
+## Special
```
Comment (emilybrick) [inline content/components/icon.mdx:54]: This section makes sense to me but do we call this Special in other instances? I wonder if it should be titled 'Specific use-cases' or 'Pre-defined icons' (pre-defined meaning?)

Comment (maximedegreve) [reply content/components/icon.mdx:54]: I had specific myself first as well so I can change it back!

**Review by emilybrick — APPROVED** [review]

**Review by josepmartins — APPROVED** [review]
